sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/ORCCarbonWriter.java

```diff
@@ -0,0 +1,173 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.carbondata.sdk.file;
+
+import java.io.IOException;
+import java.util.*;
+
+import org.apache.carbondata.core.constants.CarbonCommonConstants;
+import org.apache.carbondata.core.datastore.filesystem.CarbonFile;
+import org.apache.carbondata.sdk.file.utils.SDKUtil;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.ql.io.orc.OrcStruct;
+import org.apache.hadoop.hive.ql.io.orc.Reader;
+import org.apache.hadoop.hive.ql.io.orc.RecordReader;
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.StructObjectInspector;
+import org.apache.hadoop.io.Text;
+
+
+/**
+ * Implementation to write ORC rows in CSV format to carbondata file.
+ */
+public class ORCCarbonWriter extends CSVCarbonWriter {
+  private Configuration configuration;
+  private CSVCarbonWriter csvCarbonWriter = null;
+  private Reader orcReader = null;
+  private CarbonFile[] dataFiles;
+
+  ORCCarbonWriter(CSVCarbonWriter csvCarbonWriter, Configuration configuration) {
+    this.csvCarbonWriter = csvCarbonWriter;
+    this.configuration = configuration;
+  }
+
+  @Override
+  public void setDataFiles(CarbonFile[] dataFiles) {
+    this.dataFiles = dataFiles;
+  }
+
+  /**
+   * Load ORC file in iterative way.
+   */
+  @Override
+  public void write() throws IOException {
+    if (this.dataFiles == null || this.dataFiles.length == 0) {
+      throw new RuntimeException("'withOrcPath()' must be called to support loading ORC files");
+    }
+    if (this.csvCarbonWriter == null) {
+      throw new RuntimeException("csv carbon writer can not be null");
+    }
+    Arrays.sort(this.dataFiles, Comparator.comparing(CarbonFile::getPath));
+    for (CarbonFile dataFile : this.dataFiles) {
+      this.loadSingleFile(dataFile);
+    }
+  }
+
+  private void loadSingleFile(CarbonFile file) throws IOException {
+    orcReader = SDKUtil.buildOrcReader(file.getPath(), this.configuration);
+    ObjectInspector objectInspector = orcReader.getObjectInspector();
+    RecordReader recordReader = orcReader.rows();
+    if (objectInspector instanceof StructObjectInspector) {
+      StructObjectInspector structObjectInspector =
+          (StructObjectInspector) orcReader.getObjectInspector();
+      while (recordReader.hasNext()) {
+        Object record = recordReader.next(null); // to remove duplicacy.
+        List valueList = structObjectInspector.getStructFieldsDataAsList(record);
+        for (int i = 0; i < valueList.size(); i++) {
+          valueList.set(i, parseOrcObject(valueList.get(i), 0));
+        }
+        this.csvCarbonWriter.write(valueList.toArray());
+      }
+    } else {
+      while (recordReader.hasNext()) {
```

Comment:
Curious to know when does this else case hit ? You testcase do not seem to cover it

Reply:
This case will happen when ORC schema is not an instance of StructObjectInspector. Added the test case for the same now.